AdyenActions/Components/3DS2/Action handlers/ThreeDS2ClassicActionHandler.swift

```diff
@@ -85,27 +86,41 @@ internal class ThreeDS2ClassicActionHandler: AnyThreeDS2ActionHandler, Component
     /// - Parameter completionHandler: The completion closure.
     /// :nodoc:
     internal func handle(_ challengeAction: ThreeDS2ChallengeAction,
-                         completionHandler: @escaping (Result<ThreeDSActionHandlerResult, Error>) -> Void) {
+                         completionHandler: @escaping (Result<ThreeDSActionHandlerResult, ThreeDS2ActionHandlerError>) -> Void) {
         let event = Analytics.Event(
             component: challengeEventName,
             flavor: _isDropIn ? .dropin : .components,
             environment: apiContext.environment
         )
         coreActionHandler.handle(challengeAction, event: event) { [weak self] result in
+            guard let self = self else { return }
             switch result {
             case let .success(result):
-                self?.handle(result, completionHandler: completionHandler)
+                self.handle(result, completionHandler: completionHandler)
             case let .failure(error):
-                completionHandler(.failure(error))
+                completionHandler(.failure(self.onReceiveError(error: error)))
             }
         }
     }
 
     private func handle(_ threeDSResult: ThreeDSResult,
-                        completionHandler: @escaping (Result<ThreeDSActionHandlerResult, Error>) -> Void) {
+                        completionHandler: @escaping (Result<ThreeDSActionHandlerResult, ThreeDS2ActionHandlerError>) -> Void) {
         let additionalDetails = ThreeDS2Details.challengeResult(threeDSResult)
         completionHandler(.success(.details(additionalDetails)))
     }
+    
+    private func onReceiveError(error: ThreeDS2CoreActionHandlerError) -> ThreeDS2ActionHandlerError {
+        switch error {
+        case let .cancellationAction(threeDSResult):
+            return .cancellation(ThreeDS2Details.challengeResult(threeDSResult))
+        case .missingTransaction:
+            return .missingTransaction
+        case let .unknown(unknownError):
+            return .unknown(unknownError)
+        case let .underlyingError(underlyingError):
+            return .underlyingError(underlyingError)
+        }
```

Comment:
here you can use the `ThreeDS2ActionHandlerError` initializer you implemented, right? :-)